internal/xds/translator/route.go

```diff
@@ -15,22 +15,32 @@ import (
 	previoushost "github.com/envoyproxy/go-control-plane/envoy/extensions/retry/host/previous_hosts/v3"
 	matcherv3 "github.com/envoyproxy/go-control-plane/envoy/type/matcher/v3"
 	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/structpb"
 	"google.golang.org/protobuf/types/known/wrapperspb"
 
 	"github.com/envoyproxy/gateway/internal/ir"
 	"github.com/envoyproxy/gateway/internal/utils/protocov"
 )
 
 const (
-	retryDefaultRetryOn             = "connect-failure,refused-stream,unavailable,cancelled,retriable-status-codes"
-	retryDefaultRetriableStatusCode = 503
-	retryDefaultNumRetries          = 2
+	retryDefaultRetryOn                 = "connect-failure,refused-stream,unavailable,cancelled,retriable-status-codes"
+	retryDefaultRetriableStatusCode     = 503
+	retryDefaultNumRetries              = 2
+	envoyGatewayMetadataNamespace       = "io.envoyproxy.gateway"
```

Comment:
should this be `envoy-gateway` instead ?

Reply:
> Several parts of Envoy configuration (e.g. listeners, routes, clusters) contain a [metadata](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/base.proto#envoy-v3-api-msg-config-core-v3-metadata) where arbitrary key-value pairs can be encoded. The typical pattern is to use the filter names in reverse DNS format as the key and encode filter specific configuration metadata in the value.

https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/data_sharing_between_filters#metadata

However, I agree that:
- There is no filter in this case, so, we're not bound by these conventions. Istio uses a namespace called `istio` (and not `io.istio`) for similar purposes. 
- the long name is not convenient for access log formatters accessing the relevant keys.